tests/Unit/Evolution/EventsAndDenseTriggers/DenseTriggers/Test_Filter.cpp

```diff
@@ -0,0 +1,173 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Framework/TestingFramework.hpp"
+
+#include <array>
+#include <memory>
+#include <string>
+
+#include "DataStructures/DataBox/DataBox.hpp"
+#include "Evolution/EventsAndDenseTriggers/DenseTrigger.hpp"
+#include "Evolution/EventsAndDenseTriggers/DenseTriggers/Filter.hpp"
+#include "Framework/TestCreation.hpp"
+#include "Framework/TestHelpers.hpp"
+#include "Options/Options.hpp"
+#include "Parallel/RegisterDerivedClassesWithCharm.hpp"
+#include "ParallelAlgorithms/EventsAndTriggers/Trigger.hpp"
+#include "Utilities/Registration.hpp"
+
+namespace {
+template <typename RegistrarList>
+class TestTrigger : public Trigger<RegistrarList> {
+ public:
+  /// \cond
+  TestTrigger() = default;
+  explicit TestTrigger(CkMigrateMessage* /*unused*/) noexcept {}
+  using PUP::able::register_constructor;
+  WRAPPED_PUPable_decl_template(TestTrigger);  // NOLINT
+  /// \endcond
+
+  struct Result {
+    using type = bool;
+    constexpr static Options::String help = "Result";
+  };
+
+  using options = tmpl::list<Result>;
+  constexpr static Options::String help = "help";
+
+  explicit TestTrigger(const bool result) noexcept : result_(result) {}
+
+  using argument_tags = tmpl::list<>;
+  bool operator()() const noexcept { return result_; }
+
+  // NOLINTNEXTLINE(google-runtime-references)
+  void pup(PUP::er& p) noexcept override {
+    Trigger<RegistrarList>::pup(p);
+    p | result_;
+  }
+
+ private:
+  bool result_{};
+};
+
+/// \cond
+template <typename RegistrarList>
+PUP::able::PUP_ID TestTrigger<RegistrarList>::my_PUP_ID = 0;  // NOLINT
+/// \endcond
+
+template <typename RegistrarList>
+class TestDenseTrigger : public DenseTrigger<RegistrarList> {
```

Comment:
This seems almost identical to the `TestTrigger` from the `Or` test. Consider factoring into a helper -- tests for other combination-style triggers seem like they would also have use-cases for a similar class for testing.